### PR TITLE
Remove std::filesystem to workaround OSX features.

### PR DESCRIPTION
**Pull request by thorstenhater**
Replaces use of `std::filesystem::path` with plain old `std::string`.

### CODE REVIEW

**Comment by bcumming** [comment]
bors try

File: arbor/include/arbor/mechcat.hpp [hunk 1]
```diff
@@ -129,6 +128,6 @@ const mechanism_catalogue& global_bbp_catalogue();
 
 // Load catalogue from disk.
 
-const mechanism_catalogue& load_catalogue(const std::filesystem::path&);
+    const mechanism_catalogue& load_catalogue(const std::string&);
```
Comment (bcumming) [inline arbor/include/arbor/mechcat.hpp:131]: Remove indentation.

**Review by bcumming — APPROVED** [review]
Looks good!

One pedantic indentation comment.

File: arbor/include/arbor/mechcat.hpp (hunk 1 above)
Comment (thorstenhater) [reply arbor/include/arbor/mechcat.hpp]: Done.

**Comment by halfflat** [comment]
I'll make an issue for the proposed restd::fsification of the interface with a workaround for targets such as Mac OS 10.14, such as we discussed in chat.

**Pull request merged**